server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/HdfsSplittableDataAccessor.java

```diff
@@ -128,21 +136,34 @@ protected boolean getNextSplit() throws IOException {
      */
     @Override
     public OneRow readNextObject() throws IOException {
+        if (isFileBlob && firstLine) {
+            firstLine = false;
+            return new OneRow(null, "\"");
+        }
+
+        if (isFileBlob && lastLine) {
+            // return null after the last line
+            return null;
+        }
+
```

Comment:
this is only applicable to blobs, so maybe roll into 1 top-level if:
```suggestion
if (isFileBlob) {
    if (firstLine) {
            firstLine = false;
            return new OneRow(null, "\"");
    }
    if (lastLine) {
            // return null after the last line
            return null;
    }
}
```

Reply:
makes sense

Reply:
done